tests/httptoolkit/service/test_async_service.py

```diff
@@ -0,0 +1,242 @@
+import json
+from datetime import timedelta
+
+import httpx
+import pytest
+from pytest_httpx import HTTPXMock
+from typing import Any
+
+from httptoolkit import Header, AsyncHttpxService
+from httptoolkit.errors import HttpError, ServiceError
+from httptoolkit.service import AsyncService
+from httptoolkit.transport import AsyncHttpxTransport
+from tests.httptoolkit.conftest import CustomJSONEncoder
+
+
+class DummyAsyncService(AsyncService):
+    pass
+
+
+@pytest.fixture()
+def async_service() -> AsyncService:
+    return DummyAsyncService(
+        transport=AsyncHttpxTransport(base_url="https://kontur.ru", proxies={"http://": "http://10.10.1.10:3128"}),
+        headers=(Header(name="ServiceHeader", value="async_service-header", is_sensitive=False),),
+    )
+
+
+@pytest.fixture
+def async_service_with_custom_json_encoder() -> AsyncService:
+    return AsyncService(
+        transport=AsyncHttpxTransport(
+            base_url="https://example.com:4321",
+            json_encoder=lambda content: json.dumps(content, cls=CustomJSONEncoder),
+        ),
+    )
+
+
+@pytest.fixture
+def async_service_with_only_url() -> AsyncHttpxService:
+    return AsyncHttpxService("https://example.com:4321")
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("method", ("get", "post", "put", "delete", "patch"))
+async def test_requests_when_successful(
+    method: str,
+    async_service: AsyncService,
+    httpx_mock: HTTPXMock,
+):
+    response_text = "Schwarzenegger is a woman!"
```

Comment:
Are we sure that no one will be offended by this joke? :')

Reply:
Do you think so? 😅

Reply:
Sure :')

Reply:
removed it 😢

Reply:
so i returned back

Reply:
With the hope for a healthy humor :'D